content/en/docs/concepts/services-networking/network-policies.md

```diff
@@ -312,6 +322,56 @@ The value of the label is the namespace name.
 While NetworkPolicy cannot target a namespace by its name with some object field, you can use the
 standardized label to target a specific namespace.
 
+## Pod lifecycle
+
+When the NetworkPolicy rules should be applied in the pod's lifecycle is undefined, since
+there is a potential race if the pod and the NetworkPolicy that selects that pod are created
+at around the same time. Nevertheless, if the policy was created a sufficiently long time
+before the pod it selects, it is expected that all isolation rules should be set
+before any of pod containers are started. Thus, we ensure that a pod can never establish 
+an outbound connection or receive an inbound connection that is not allowed by NetworkPolicy 
+at any point in its lifecycle. At the same time allow rules may be installed later and pod 
+should be able to handle some delay before its network is fully setup.
+
+Network plugins may implement extra `status.conditions` for pod readiness based on the
+status of network policies. These conditions then may be used by 
+[Pod readiness gate](/docs/concepts/workloads/pods/pod-lifecycle/#pod-readiness-gate).
+
+## `hostNetwork` pods
```

Comment:
I didn't include `NetworkPolicy and` part, since the whole document is about network policies, so I am not sure if it adds value. 
Second-level heading would be nice, but I guess we need to find a set of headers that would go under it, otherwise it is difficult to figure out the right heading name if there will be only 1 sub-section. Maybe you have an idea of what else could go under a new second-level header?

Reply:
I am not sure I understand how exactly that is supposed to work, and I since this PR already covers 3 new sections, maybe we better document that limitation in another PR?

Reply:
sure I am happy to improve it, but I can't figure out a better name. Updated it to "NetworkPolicy and `hostNetwork` pods" for now.